### PR TITLE
Strip whitespace before adding new users to the database

### PR DESCRIPTION
**Pull request by avinhurry**
### Context

Users were being added with unintentional leading and proceeding white space, causing login issues.

### Changes proposed in this pull request

Strip the white space before and after the email address.

### Guidance to review

[Test it out here](https://publish-review-4112.test.teacherservices.cloud/support/2024/users/new)

### Checklist

- [x] Make sure all information from the Trello card is in here
- [x] Attach to Trello card
- [x] Rebased main
- [x] Cleaned commit history
- [x] Tested by running locally
- [x] Inform data insights team due to database changes

### CODE REVIEW

File: app/controllers/support/users_controller.rb [hunk 1]
```diff
@@ -22,6 +22,8 @@ def edit
 
     def create
       @user = User.new(user_params)
+      @user.email.strip!
```
Comment (elceebee) [inline app/controllers/support/users_controller.rb:25]: There is a `downcase_email` in a `before_save` hook on the `User` model. My preference would be to have both the strip and the downcase happen at the same time, and on the model. There may be other methods for creating a user and we will always want the whitespace removed. 

What do you think?

Comment (elceebee) [reply app/controllers/support/users_controller.rb:25]: Sweet!

Comment (avinhurry) [reply app/controllers/support/users_controller.rb]: Yup defo agree, updated here: https://github.com/DFE-Digital/publish-teacher-training/pull/4112/commits/3661e071d72c5723c631c94002bdb93dd52d2711

File: app/models/user.rb [hunk 2]
```diff
@@ -5,7 +5,7 @@ class User < ApplicationRecord
   include PgSearch::Model
   include RecruitmentCycleHelper
 
-  before_save :downcase_email
+  before_save :downcase_email, :strip_email_whitespace
```
Comment (elceebee) [inline app/models/user.rb:8]:
```suggestion
  normalizes :email, with: -> email { email.strip.downcase }
```

That should work without the `before_save` hook, according the documentation.

Comment (avinhurry) [reply app/models/user.rb:8]: This only possible in rails 7.1 **or** with the normalize gem. I don't think it's worth adding the gem just for this. 

https://www.shakacode.com/blog/rails-add-normalize-to-activerecord/#:~:text=Normalization%20is%20applied%20when%20the,subsequent%20querying%20using%20unnormalized%20values.

Comment (elceebee) [reply app/models/user.rb:8]: Ooh right! I hadn't realised we weren't on Rails 7 yet. Sorry about that!

**Review by elceebee — APPROVED** [review]

File: app/models/user.rb (hunk 2 above)
Comment (avinhurry) [reply app/models/user.rb:8]: No probs, we need to do the upgrade!

**Pull request merged**